include/rhash.h

```diff
@@ -105,11 +105,14 @@ extern "C" {
   /* closes the track handle */
   typedef void (*rc_hash_cdreader_close_track_handler)(void* track_handle);
 
+  typedef uint32_t(*rc_hash_cdreader_get_lba_handler)(void* track_handle);
+
   struct rc_hash_cdreader
   {
     rc_hash_cdreader_open_track_handler      open_track;
     rc_hash_cdreader_read_sector_handler     read_sector;
     rc_hash_cdreader_close_track_handler     close_track;
+    rc_hash_cdreader_get_lba_handler         get_lba;
```

Comment:
I really don't like this implementation. `get_lba` is retrieving supplemental data captured by open_track. I understand the need for it based on stepping through the code, but I think a cleaner solution is to encode the `lba` in `cdrom->first_sector_offset` as follows:
```
cdrom->first_sector_offset = -(lba * cdrom->sector_size);
```
This doesn't completely solve the problem though. `rc_cd_find_file_sector` would no longer be able to find the TOC data in sector 16, so you'd have to modify it to take a TOC sector parameter that would be set to 45016 for Dreamcast. This solution would keep the "CD" interface clean and abstract the gdi-specific behavior to the Dreamcast hashing code.

I also recommend using an overloaded method instead of changing all the calls. An extra parameter of 0 (or 16 in this case) has very little meaning to people reading the code. Hiding it in a helper function makes the code more readable.
```
static uint32_t rc_cd_find_file_sector_toc(void* track_handle, const char* path, unsigned* size, 
                                           unsigned toc_sector)
{
    .. existing function, modified to use toc_sector parameter
}

static uint32_t rc_cd_find_file_sector(void* track_handle, const char* path, unsigned* size)
{
    /* the cd information is usually 16 frames in */
    return rc_cd_find_file_sector_toc(track_handle, path, size, 16);
}
``` 
The resulting sector would then automatically be adjusted to the `last_track_handle`'s `first_sector_offset`, finding the binary without having to remember to make another adjustment based on the LBA of the last track.

Reply:
Your "really high number" is a negative number (-105,840,000), and is effectively removing the 45000 sectors from the directory entries. As I mentioned, the one caveat to this is the TOC sector (that's normally in sector 16) has to be offset in the other direction. The 16th sector of track 3 is sector 45016, so that's where the TOC should be read from. If you look for the TOC in sector 45016, the calculation becomes:
```
sector_start = 45016 * 2352 + 16 + (-105840000)
sector_start = 105877632 + 16 - 105840000
sector_start = 37648
```
resulting in a positive offset in the track 3 file adjusted for the lba.

Similarly, when the TOC says the boot file is in sector 45020, sector_start comes out to 47056.

The meta information can be similarly located at sector 45000.

Reply:
I understand now, that's a pretty good solution!
I've now just commited all changes mentioned in the code review.